MH4U Dex/CoreDataController.m

```diff
@@ -0,0 +1,430 @@
+
+@implementation CoreDataController
+
+#pragma mark - Initial Data Loading
+
+- (void)loadMonsterData
+{
+    // Because monsters are added first, there are no relationships to set up, so we can hydrate them.
+    NSString *path = [[NSBundle mainBundle] pathForResource:@"monsters" ofType:@"json"];
+    NSDictionary *attributes = @{@"id":@"_id", @"monster_class":@"class", @"name":@"name", @"jpn_name":@"jpn_name", @"sort_name":@"sort_name", @"trait":@"trait", @"icon":@"icon_name", @"url":@"url"};
+    [self.managedObjectContext hydrateStoreWithJSONAtPath:path attributeMappings:attributes forEntityName:@"Monster"];
+    path = [[NSBundle mainBundle] pathForResource:@"damageZones" ofType:@"json"];
+    attributes = @{@"id":@"_id", @"monster_id":@"monster_id", @"monsterName":@"monster_name", @"jpnMonsterName":@"jpn_monster_name", @"bodyPart":@"body_part", @"cut":@"cut", @"impact":@"impact", @"shot":@"shot", @"fire":@"fire", @"water":@"water", @"thunder":@"thunder", @"ice":@"ice", @"dragon":@"dragon", @"ko":@"ko", @"extract":@"extract"};
+    [self.managedObjectContext hydrateStoreWithJSONAtPath:path attributeMappings:attributes forEntityName:@"DamageZone"];
+}
+
+- (void)loadMonsterDropData
+{
+    NSArray *dropList = [self loadArrayFromJsonFileNamed:@"monsterDrops"];
+    for (NSDictionary *drop in dropList) {
+        // Only add the drop if it is not already there
+        if (![self checkCountForEntityWithEntityName:@"MonsterDrop" withIDNumber:drop[@"_dropID"]]) {
+            NSEntityDescription *newDropDescription = [NSEntityDescription entityForName:@"MonsterDrop" inManagedObjectContext:self.managedObjectContext];
+            MonsterDrop *newDrop = [[MonsterDrop alloc] initWithEntity:newDropDescription insertIntoManagedObjectContext:self.managedObjectContext];
+            [newDrop setDropBasicValuesWithID:drop[@"_dropID"] method:drop[@"method"] quantity:drop[@"quantity"]];
+            
+            // If the Item for this drop already exists in the persistent store ...
+            Item *item = (Item *)[self getUniqueEntityWithEntityName:@"Item" withNameAttribute:drop[@"item_name"]];
+            if (item) {
+                // We need to hook up relationships appropriately.
+                newDrop.item = item;
+                [item.monsterSourceSet addObject:newDrop];
+            } else {
+                // Otherwise, we should create the item.
+                NSEntityDescription *newItemDescription = [NSEntityDescription entityForName:@"Item" inManagedObjectContext:self.managedObjectContext];
+                Item *newItem = [[Item alloc] initWithEntity:newItemDescription insertIntoManagedObjectContext:self.managedObjectContext];
+                [newItem.monsterSourceSet addObject:newDrop];
+                newItem.name = drop[@"item_name"];
+                newDrop.item = newItem;
+            }
+            
+            // If this is a monster drop ...
+            if (![drop[@"monster_name"] isEqualToString:@""]) {
```

Comment:
If `drop` or `drop[@"monster_name"]` is `nil`, then `[drop[@"monster_name"] isEqualToString:@""]` will be `nil`, and `![drop[@"monster_name"] isEqualToString:@""]` will be true—is that intended?  Otherwise, you might want something like `drop[@"monster_name"].length`, which will be true whenever `drop[@"monster_name"]` is not `nil` and also has a nonzero length.  You might also want to verify that `drop[@"monster_name"]` is actually an `NSString`, but that may already be guaranteed in the context of this method.